auxx/src/Lang/Parser.hs

```diff
@@ -66,16 +67,14 @@ gExpr = mdo
         ] <?> "atom"
     return ntExpr
 
-pExpr :: Parser Text [Token] Expr
+pExpr :: Parser Text [(s, Token)] Expr
 pExpr = parser gExpr
 
-newtype ParseError = ParseError (Report Text [Token])
+data ParseError = ParseError Text (Report Text [(Span, Token)])
```

Comment:
Can you use records here please? I have no idea what `Text` denotes by looking at this type definition.

(it becomes clear if I look below, but I believe it should be clear by just looking at the definition)